src/Illuminate/Routing/Router.php

```diff
@@ -251,13 +251,29 @@ public function redirect($uri, $destination, $status = 301)
      * @param  array  $data
      * @return \Illuminate\Routing\Route
      */
-    public function view($uri, $view, $data = [])
+    public function view($uri, $view = null, $data = [])
     {
+        if (func_num_args() == 2 && is_array($view)) {
+            $data = $view;
+            $view = null;
+        }
+
         return $this->match(['GET', 'HEAD'], $uri, '\Illuminate\Routing\ViewController')
-                ->defaults('view', $view)
+                ->defaults('view', $view ?: $this->toViewFormat($uri))
```

Comment:
I'm not sure if this needs to be separated out into its own method, seems like something un-necessary for something simple?

Reply:
I added as a own method because I could have forgot some rule on the conversion of URI to view string